src/dbcp/data_mart/projects.py

```diff
@@ -253,25 +245,19 @@ def _get_proprietary_proposed_offshore(engine: sa.engine.Engine) -> pd.DataFrame
     """
     query = """
     WITH
-    cable_locs as (
+    proj_county_assoc as (
         SELECT
             project_id,
             locs.county_id_fips,
-            COUNT(*) OVER(PARTITION BY project_id) AS n_locations
+            -- Note that "frac_locations_in_county" is a misnomer. It is actually
+            -- "fraction_of_locations_represented_by_this_row". When I originally
+            -- named it, I thought location:county was m:1, but it's actually m:m
+            -- because some projects list multiple towns in the same parent county
+            -- in the raw "county" field.
+            (1.0 / count(*) over (partition by project_id))::real as frac_locations_in_county
         FROM data_warehouse.offshore_wind_cable_landing_association as cable
         INNER JOIN data_warehouse.offshore_wind_locations as locs
         USING(location_id)
-    ),
-    proj_county_assoc as (
-        SELECT
-            project_id,
-            county_id_fips,
-            -- some counties have multiple cable landings from the same
-            -- project (different towns). I allocate the capacity equally
-            -- over the landings
-            (count(*) * 1.0 / max(n_locations))::real as frac_locations_in_county
-        FROM cable_locs
-        group by 1,2
```

Comment:
Is this the groupby step your PR description is referring to? So is this allocation over multiple counties not happening any more or is it done in a different place now?

Reply:
Ya. The previous version simplified things by combining multiple towns in the same county (only 12 such cases) to a single record. For example, if a raw project looked like:

| project_id | raw_county | geocoded_county |
| --- | --- | --- |
| 1 | County 1 | County 1 |
| 1 | Town 1 | County 2 |
| 1 | Town 2 | County 2 |

The old output would look like:

| project_id | geocoded_county | frac_locations_in_county |
| --- |  --- | --- |
| 1 | County 1 | 0.333 |
| 1 | County 2 | 0.667 |

The new output would be:

| project_id | geocoded_county | frac_locations_in_county |
| --- |  --- | --- |
| 1 | County 1 | 0.333 |
| 1 | County 2 | 0.333 |
| 1 | County 2 | 0.333 |

LBNL was already handled in the same way as this new method (for its 8 multi-county cases), but with a drop_duplicates after, which only took the first record. So everything was inconsistent.

The downstream impact is small, mostly because this is a rare thing (only 8 in LBNL and 12 in offshore). The county aggregate MW figures won't change, but project counts will increase for the handful of affected counties.

Reply:
Got it thanks!